torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -720,14 +723,24 @@ def rewrite_node(node):
                 return g.call_function(ttnn.split, args=new_args)
 
             if node.target == torch.ops.aten._to_copy.default:
+                # Keep it if casting to bool type(bool may be problematic)
+                if kwargs["dtype"] in [torch.bool]:
+                    return None
+                # Keep it if the graph output uses this op
                 target_users_ops = [user.target for user in node.users.keys()]
-                # Float and int types can be converted to ttnn.bfloat16, but bool may be problematic
-                # Skip if type casting from bool and if the graph output uses this op
-                if kwargs["dtype"] not in [torch.bool] and "output" not in target_users_ops:
-                    # Essentially remove this op because it's used as a typecast
-                    return node.args[0]
-                else:
+                if "output" in target_users_ops:
                     return None
+                src_dtype = node.args[0].meta["val"].dtype
+                dst_dtype = kwargs["dtype"]
+                # Some aten op need it to cast specific dtype (ex, index_select)
+                # Keep it if casting from int to float or reverse
+                if dst_dtype in [torch.int32, torch.int64] and src_dtype not in [torch.int32, torch.int64]:
+                    return None
+                if src_dtype in [torch.int32, torch.int64] and dst_dtype not in [torch.int32, torch.int64]:
+                    return None
+                target_users_ops = [user.target for user in node.users.keys()]
```

Comment:
This line probably not needed?

Reply:
I had removed it.

I wanted to avoid adding `aten._to_copy` to the user's output at the beginning. Because including it would mean that the model would always have at least one `aten._to_copy` operation, so that cannot achieve end to end compilation.

However, Falcon uses multiple graphs, and in some cases, one graph's output serves as the third index selection for another graph. In these cases, `aten._to_copy` is still needed to preserve the dtype.